java/core/src/main/java/com/microsoft/bond/Throw.java

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import java.util.Locale;
+
+/**
+ * Encapsulates Bond exception throwing and error messages.
+ * Some message details (e.g. list indices, map keys) may be added to
+ * exception message by certain specialized implementations of Bond type.
+ */
+final class Throw {
+
+    // invariant locale
+    private static final Locale LOCALE = Locale.ROOT;
+
+    // general entry point to raise a data exception on serialization or deserialization
+    static void raiseInvalidDataError(
+            Exception cause,
+            String format,
+            Object... args) throws InvalidBondDataException {
+        String message = String.format(LOCALE, format, args);
+        throw new InvalidBondDataException(message, cause);
+    }
+
+    // general entry point to raise a data exception on value comparison
+    static void raiseIllegalArgumentException(
```

Comment:
Removed this method, it was remnant of older functionality I decided to exclude for now.

---
In reply to: [123380136](https://github.com/Microsoft/bond/pull/502#discussion_r123380136) [](ancestors = 123380136)